endpoints/openrtb2/auction_test.go

```diff
@@ -1145,6 +1145,135 @@ func TestStoredRequests(t *testing.T) {
+		},
+	}
+
+	for _, test := range testCases {
+		req := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(test.givenBidRequestData))
+
+		newReq, _, errL := deps.parseRequest(req)
+		assert.Empty(t, errL, test.description)
+
+		requestExt, err := newReq.GetRequestExt()
+		assert.Empty(t, err, test.description)
+
+		requestPrebid := requestExt.GetPrebid()
+		assert.Equalf(t, test.expectedChannelObject, requestPrebid.Channel, "Channel information isn't correct: %s\n", test.description)
+	}
+}
+
+func TestValidateOrFillChannel(t *testing.T) {
+	testCases := []struct {
+		description            string
+		givenBidRequestData    string
+		givenIsAmp             bool
+		expectedError          error
+		expectedChannelObject  *openrtb_ext.ExtRequestPrebidChannel
+		expectedAmpChannelInfo json.RawMessage
+	}{
+		{
+			description:           "No channel object in app request, so we expect channel name to be set to app",
+			givenBidRequestData:   testBidRequestForChannelTest[0],
+			givenIsAmp:            false,
+			expectedError:         nil,
+			expectedChannelObject: &openrtb_ext.ExtRequestPrebidChannel{Name: "app", Version: ""},
+		},
+		{
+			description:           "Channel object in request with populated name/version, we expect same name/version in object that's created",
+			givenBidRequestData:   testBidRequestForChannelTest[1],
+			givenIsAmp:            false,
+			expectedError:         nil,
+			expectedChannelObject: &openrtb_ext.ExtRequestPrebidChannel{Name: "video", Version: "1.0"},
+		},
+		{
+			description:           "No channel object in site request, expect nil",
+			givenBidRequestData:   testBidRequestForChannelTest[2],
+			givenIsAmp:            false,
+			expectedError:         nil,
+			expectedChannelObject: nil,
+		},
+		{
+			description:           "No prebid object, expect app channel object to be created",
+			givenBidRequestData:   testBidRequestForChannelTest[3],
+			givenIsAmp:            false,
+			expectedError:         nil,
+			expectedChannelObject: &openrtb_ext.ExtRequestPrebidChannel{Name: "app", Version: ""},
+		},
+		{
+			description:            "No prebid object, expect amp channel info to be generated",
+			givenBidRequestData:    testBidRequestForChannelTest[3],
+			givenIsAmp:             true,
+			expectedError:          nil,
+			expectedAmpChannelInfo: json.RawMessage(`{"prebid":{"channel":{"name":"amp","version":""}}}`),
+		},
+		{
+			description:           "No channel name given in channel object, we expect error to be thrown",
+			givenBidRequestData:   testBidRequestForChannelTest[4],
+			givenIsAmp:            false,
+			expectedError:         errors.New("ext.prebid.channel.name can't be empty"),
+			expectedChannelObject: nil,
+		},
+	}
+
+	for _, test := range testCases {
+		req := &openrtb_ext.RequestWrapper{}
+		req.BidRequest = &openrtb2.BidRequest{}
+
+		err := json.Unmarshal([]byte(test.givenBidRequestData), req.BidRequest)
+		assert.Empty(t, err, test.description)
```

Comment:
The `validateOrFillChannel` expects the request being passed in as an object. Is there a reason why we're specifying the requests in JSON and then unmarshalling them before passing them to the request? Also for unit tests, I'd suggest keep the requests as small as possible by specifying only fields that are needed to test the function under consideration. For example, you can modify these test cases to look something like:

```
{
	description: "App request; No request.ext present; expect app channel info to be generated",
	givenReqWrapper: &openrtb_ext.RequestWrapper{
		BidRequest: &openrtb2.BidRequest{App: &openrtb2.App{}},
	},
	givenIsAmp:         false,
	expectedError:      nil,
	expectedRequestExt: []byte(`{"prebid":{"channel": {"name": "app", "version": ""}}}`),
},
```

Reply:
That makes a lot of sense, I'll use this approach instead!